### PR TITLE
feat: Add zcap in create data vault

### LINKED ISSUE #142: Add zcap in create data vault

### LINKED ISSUE #143: Setup local kms

### PR DESCRIPTION
**Pull request by fqutishat**
closes #142 

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/auth/zcapld/service.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	if err != nil {
+		return nil, fmt.Errorf("failed to open store %s: %w", storeName, err)
+	}
+
+	return &Service{keyManager: keyManager, crypto: crypto, store: store}, nil
+}
+
+// Create zcap payload
+func (s *Service) Create() ([]byte, error) {
+	rootCapability, err := s.createRootCapability()
```
Comment (llorllale) [inline pkg/auth/zcapld/service.go:47]: I should have made this clearer: there should be one root capability per resource.

So if a user creates Vault ID `12345` then the service needs to create a root capability _for that resource_ and store it. Then it should create a new zcap pointing to `12345` as the parent and with allowed actions.

Comment (llorllale) [reply pkg/auth/zcapld/service.go:47]: The invocation target of both the root capability and the child capability should identify the resource.

Comment (fqutishat) [reply pkg/auth/zcapld/service.go:47]: I'm creating one root capability per Vault ID

Comment (llorllale) [reply pkg/auth/zcapld/service.go:47]: How are they specifying the vault ID as the invocation target? I don't see it.

Comment (fqutishat) [reply pkg/auth/zcapld/service.go:47]: @llorllale the flow you explained it's there. Create method will be called from create Vault ID logic.

Comment (fqutishat) [reply pkg/auth/zcapld/service.go]: Done

**Review by rolsonquadras — APPROVED** [review]

**Review by Baha-sk — APPROVED** [review]

**Review by llorllale — APPROVED** [review]

**Review by sandrask — APPROVED** [review]

**Pull request merged**